core/templates/pages/topics-and-skills-dashboard-page/topics-list/topics-list.directive.spec.ts

```diff
@@ -0,0 +1,149 @@
+  var $rootScope = null;
+  var directive = null;
+  var AlertsService = null;
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(angular.mock.inject(function($injector) {
+    $uibModal = $injector.get('$uibModal');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $httpBackend = $injector.get('$httpBackend');
+    directive = $injector.get('topicsListDirective')[0];
+    $q = $injector.get('$q');
+    AlertsService = $injector.get('AlertsService');
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope,
+      $uibModal,
+    });
+  }));
+
+  it('should init the controller', function() {
+    ctrl.$onInit();
+    const topicHeadings = [
+      'index', 'name', 'canonical_story_count', 'subtopic_count',
+      'skill_count', 'topic_status'
+    ];
+    expect(ctrl.selectedIndex).toEqual(null);
+    expect(ctrl.TOPIC_HEADINGS).toEqual(topicHeadings);
+  });
+
+  it('should return topic editor url', function() {
+    const topicId1 = 'uXcdsad3f42';
+    const topicId2 = 'aEdf44DGfre';
+    expect(ctrl.getTopicEditorUrl(topicId1)).toEqual(
+      '/topic_editor/uXcdsad3f42');
+    expect(ctrl.getTopicEditorUrl(topicId2)).toEqual(
+      '/topic_editor/aEdf44DGfre');
+  });
+
+  it('should select and show edit options for a topic', function() {
+    const topicId1 = 'uXcdsad3f42';
+    const topicId2 = 'aEdf44DGfre';
+    expect(ctrl.showEditOptions(topicId1)).toEqual(false);
+    expect(ctrl.showEditOptions(topicId2)).toEqual(false);
+
+    ctrl.enableEditOptions(topicId1);
+    expect(ctrl.showEditOptions(topicId1)).toEqual(true);
+    expect(ctrl.showEditOptions(topicId2)).toEqual(false);
+
+    ctrl.enableEditOptions(topicId2);
+    expect(ctrl.showEditOptions(topicId1)).toEqual(false);
+    expect(ctrl.showEditOptions(topicId2)).toEqual(true);
+
+    ctrl.enableEditOptions(null);
+    expect(ctrl.showEditOptions(topicId1)).toEqual(false);
+    expect(ctrl.showEditOptions(topicId2)).toEqual(false);
+  });
+
+  it('should open the delete topic modal', function() {
+    var modalSpy = spyOn($uibModal, 'open').and.callThrough();
+    ctrl.deleteTopic('dskfm4');
+    expect(modalSpy).toHaveBeenCalled();
+  });
+
+  it('should return serial number for topic', function() {
+    ctrl.getPageNumber = function() {
+      return 0;
+    };
+    ctrl.getItemsPerPage = function() {
+      return 10;
+    };
+    expect(ctrl.getSerialNumberForTopic(2)).toEqual(3);
+    ctrl.getPageNumber = function() {
+      return 3;
+    };
+    ctrl.getItemsPerPage = function() {
+      return 15;
+    };
+    expect(ctrl.getSerialNumberForTopic(2)).toEqual(48);
+  });
+
+  it('should delete the topic success', function() {
```

Comment:
Please make sure these read as an actual English sentence. "It should reinitialize the page after successfully deleting the topic", "It should show an error if the backend fails to delete the topic".

Ditto below, and for all tests you write in this PR and in the future.

Reply:
Updated this.